lib/DxcSupport/dxcmem.cpp

```diff
@@ -22,19 +22,9 @@
 static llvm::sys::ThreadLocal<IMalloc> *g_ThreadMallocTls;
 static IMalloc *g_pDefaultMalloc;
 
-// Used by DllMain to set up and tear down per-thread tracking.
-HRESULT DxcInitThreadMalloc() throw();
```

Comment:
So all of these function declarations are useless on Windows?

Reply:
They are duplicate prototypes of the Global.h header. I can't think of any reason why they'd be needed anywhere. This change compiles and runs on Windows, Linux, and Mac for all tests I know of. I don't know why they were there to begin with, but they have the same comments and whitespace as the Global.h header. So perhaps it was an inadvertent copy and paste.

There were two functions,  `DxcThreadAlloc`, and  `DxcThreadFree` that were declared in both places, but defined nowhere. So I removed them from Global.h as well. 

I made an earlier change that added the `throw()` attribute to each of these bodies that was missing before. Perhaps this was a way to address that issue that worked on Windows but not anywhere else? At any rate, reasoning and testing gives no indication these duplicate prototypes are necessary anymore if they ever were.

Reply:
SG. Thanks